torch/csrc/QEngine.cpp

```diff
@@ -0,0 +1,79 @@
+#include <torch/csrc/QEngine.h>
+
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/utils/object_ptr.h>
+#include <torch/csrc/utils/python_strings.h>
+
+#include <c10/core/QEngine.h>
+
+#include <structmember.h>
+#include <cstring>
+#include <string>
+
+PyObject* THPQEngine_New(at::QEngine qengine, const std::string& name) {
+  auto type = (PyTypeObject*)&THPQEngineType;
+  auto self = THPObjectPtr{type->tp_alloc(type, 0)};
+  if (!self)
+    throw python_error();
+  auto self_ = reinterpret_cast<THPQEngine*>(self.get());
+  self_->qengine = qengine;
+  std::strncpy(self_->name, name.c_str(), QENGINE_NAME_LEN);
+  self_->name[QENGINE_NAME_LEN] = '\0';
+  return self.release();
+}
+
+PyObject* THPQEngine_repr(THPQEngine* self) {
+  std::string name = self->name;
+  return THPUtils_packString("torch." + name);
+}
+
+PyTypeObject THPQEngineType = {
```

Comment:
@dzhulgakov mentioned offline that there is better way to do this type of binding. But it can be improved in separate diff.